Can't view logs of local stack if there are yet-to-be-deleted resources

In my case, the update created a new LaunchConfiguration but hasn't deleted the old one yet.

```
$ pulumi logs
Collecting logs since 2018-02-07T16:47:14.000-08:00.

panic: fatal: An assertion has failed: Unexpected duplicate resource urn:pulumi:asg-and-ecs::pulumi-template::aws:ec2/launchConfiguration:LaunchConfiguration::launchConfiguration

goroutine 1 [running]:
github.com/pulumi/pulumi/pkg/util/contract.failfast(0xc42001c8f0, 0xa4)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/util/contract/failfast.go:11 +0xf3
github.com/pulumi/pulumi/pkg/util/contract.Assertf(0x1899c00, 0x19ca565, 0x20, 0xc42018b888, 0x1, 0x1)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/util/contract/assert.go:21 +0x13d
github.com/pulumi/pulumi/pkg/operations.makeResourceTreeMap(0xc420242a00, 0xf, 0x10, 0x0, 0xffffffffffffffff)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/operations/resources.go:47 +0x207
github.com/pulumi/pulumi/pkg/operations.NewResourceTree(0xc420242a00, 0xf, 0x10, 0xc4202f2b70)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/operations/resources.go:32 +0x3f
github.com/pulumi/pulumi/pkg/backend/local.(*localBackend).GetLogs(0xc420207a10, 0xc42007a528, 0xb, 0xc4202a77a0, 0x0, 0x0, 0x44, 0xc42018ba70, 0x1013068, 0x30, ...)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/backend/local/backend.go:250 +0xee
github.com/pulumi/pulumi/pkg/backend.GetStackLogs(0x1ec2f80, 0xc420208bc0, 0xc4202a77a0, 0x0, 0x0, 0x1, 0x1, 0x0, 0x0, 0x0)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/backend/stack.go:65 +0x9e
github.com/pulumi/pulumi/pkg/backend/local.(*localStack).GetLogs(0xc420208bc0, 0xc4202a77a0, 0x0, 0x0, 0xc4202bacc0, 0x44, 0x0, 0x0, 0x31)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/backend/local/stack.go:68 +0x59
github.com/pulumi/pulumi/cmd.newLogsCmd.func1(0xc42022b8c0, 0x1f417d8, 0x0, 0x0, 0x0, 0x0)
	/Users/matthew/go/src/github.com/pulumi/pulumi/cmd/logs.go:64 +0x606
github.com/pulumi/pulumi/pkg/util/cmdutil.RunFunc.func1(0xc42022b8c0, 0x1f417d8, 0x0, 0x0)
	/Users/matthew/go/src/github.com/pulumi/pulumi/pkg/util/cmdutil/exit.go:55 +0x54
github.com/pulumi/pulumi/vendor/github.com/spf13/cobra.(*Command).execute(0xc42022b8c0, 0x1f417d8, 0x0, 0x0, 0xc42022b8c0, 0x1f417d8)
	/Users/matthew/go/src/github.com/pulumi/pulumi/vendor/github.com/spf13/cobra/command.go:702 +0x2c6
github.com/pulumi/pulumi/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4201fefc0, 0xc420015c31, 0xc420015c30, 0xc420207110)
	/Users/matthew/go/src/github.com/pulumi/pulumi/vendor/github.com/spf13/cobra/command.go:783 +0x30e
github.com/pulumi/pulumi/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4201fefc0, 0x1006424, 0xc42007a058)
	/Users/matthew/go/src/github.com/pulumi/pulumi/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/Users/matthew/go/src/github.com/pulumi/pulumi/main.go:14 +0x2b
```

--- a/pkg/operations/resources.go
+++ b/pkg/operations/resources.go
@@ -44,12 +44,15 @@ func makeResourceTreeMap(source []*resource.State) (*Resource, map[resource.URN]
 	for _, state := range source {
 		ns = state.URN.Namespace()
 		alloc = state.URN.Alloc()
-		contract.Assertf(resources[state.URN] == nil, "Unexpected duplicate resource %s", state.URN)
-		resources[state.URN] = &Resource{
-			NS:       ns,
-			Alloc:    alloc,
-			State:    state,
-			Children: make(map[resource.URN]*Resource),
+		if !state.Delete {
+			// Only include resources which are not marked as pending-deletion.
+			contract.Assertf(resources[state.URN] == nil, "Unexpected duplicate resource %s", state.URN)
+			resources[state.URN] = &Resource{
+				NS:       ns,
+				Alloc:    alloc,
+				State:    state,
+				Children: make(map[resource.URN]*Resource),
+			}
 		}
 	}
 

--- a/pkg/operations/testdata/todo.json
+++ b/pkg/operations/testdata/todo.json
@@ -1144,6 +1144,28 @@
                     ],
                     "url": "https://eupwl7wu4i.execute-api.us-east-2.amazonaws.com/"
                 }
+            },
+            {
+                "urn": "urn:pulumi:foo::todo::aws:apigateway/deployment:Deployment::todo_f569e86a",
+                "delete": true,
+                "custom": true,
+                "id": "abc123",
+                "type": "aws:apigateway/deployment:Deployment",
+                "parent": "urn:pulumi:foo::todo::cloud:http:HttpEndpoint::todo",
+                "inputs": {
+                    "description": "Deployment of version todo_f569e86a",
+                    "restApi": "eupwl7wu4i",
+                    "stageName": ""
+                },
+                "outputs": {
+                    "createdDate": "2017-11-08T19:59:03Z",
+                    "description": "Deployment of version todo_f569e86a",
+                    "executionArn": "arn:aws:execute-api:us-east-2:153052954103:eupwl7wu4i/",
+                    "id": "abc123",
+                    "invokeUrl": "https://eupwl7wu4i.execute-api.us-east-2.amazonaws.com/",
+                    "restApi": "eupwl7wu4i",
+                    "stageName": ""
+                }
             }
         ]
     }

